File: org.opencompare/api-java/src/main/java/org/opencompare/api/java/io/PCMExporter.java
package org.opencompare.api.java.io;

import org.opencompare.api.java.PCM;
import org.opencompare.api.java.PCMContainer;
import org.opencompare.api.java.PCMMetadata;

/**
 * Created by gbecan on 30/01/15.
 */
public interface PCMExporter {

    /**
     * Export PCM from a PCM container
     * @param container container to export
     * @return string representing the PCM
     */
    String export(PCMContainer container);

}

File: org.opencompare/api-scala/src/main/scala/org/opencompare/api/scala/io/PCMExporter.scala
package org.opencompare.api.scala.io

import org.opencompare.api.scala.PCM

trait PCMExporter {

  /**
    * Export a PCM into a specific format
    * @param pcm PCM to export
    * @return string representing the PCM
    */
  def export(pcm: PCM): String

}
